packages/keyring/src/keyring.ts

```diff
@@ -296,4 +296,14 @@ export class Keyring implements KeyringInstance {
   public toJson (address: string | Uint8Array, passphrase?: string): KeyringPair$Json {
     return this.#pairs.get(address).toJson(passphrase);
   }
+
+  /**
+   * @name encrypt
+   * @summary Encrypt a message using the given publickey
+   * @description Returns the encrypted message of the given message using the public key.
+   * The encrypted message can be decrypted by the corresponding keypair using keypair.decrypt() method
+   */
+  public encrypt (message: string | Uint8Array, recipientPublicKey: string | Uint8Array, recipientKeyType?: KeypairType): Uint8Array {
+    return cryptoEncrypt(message, recipientPublicKey, recipientKeyType || this.type);
```

Comment:
Generally users have no idea even what their own keypair types are, let alone somebody else knowing what a specific user has.

This means that the addition of the `recipientKeyType` here is an issue - the sender won't know which type the actual recipient uses. It could be anything and for most users this is not known at all.

Reply:
Hi @jacogr,
Thanks for your review.
I actually did not catch your point here, the users of this library are not end users, I guess that they are more techy (as  developers) and aware of what key types and encryption are.
I agree that passing the wrong key type for a given public key will fail, but we need this information.